docs/reference/MIGraphX-dev-env-vars.rst

```diff
@@ -0,0 +1,379 @@
+.. meta::
+  :description: MIGraphX environment variables for developers
+  :keywords: MIGraphX, code base, contribution, developing, env vars, environment variables
+
+========================================================
+MIGraphX environment variables
+========================================================
+
+The MIGraphX environment variables can be used by contributors to the MIGraphX code base to customize tuning, verification, and tracing.
+
+
+Model performance tunable variables
+************************************
+
+Model performance tunable variables change the compilation behavior of a model.
+ 
+.. list-table:: 
+  :widths: 30 70
+  :header-rows: 1
+
+  * - Environment variable
+    - Description
+  
+  * - ``MIGRAPHX_ENABLE_NHWC``
+    - | When set to ``1``, forces the model to use the NHWC layout.
+      | ``0``: Does not force the use of the NHWC layout.
+      | ``1``: Forces the use of the NHWC layout.
+      | Default: ``0``
+
+  * - ``MIGRAPHX_DISABLE_MLIR``
+    - | When set to ``1``, disables the use of the rocMLIR library.
+      | ``0``: Use of the rocMLIR library is enabled.
+      | ``1``: Use of the rocMLIR library is disabled. 
+      | Default: 0
+    
+
+  * - ``MIGRAPHX_ENABLE_CK``
+    - | Enables the use of the Composable Kernels library when set to 1. 
+      | Use with ``MIGRAPHX_DISABLE_MLIR``=1.
+      | ``0``: Use of the Composable Kernel libraries is disabled.
+      | ``1``: Use of the Composable Kernel libraries is enabled.
+      | Default: ``0``
+
+  * - ``MIGRAPHX_SET_GEMM_PROVIDER``
+    - Sets the GEMM provider.
+      | ``hipblaslt``: hipBLASLt is set as the GEMM provider.
+      | ``rocblas``: rocBLAS is set as the GEMM provider.
+      | Default: ``rocblas`` on gfx90a; ``hipblaslt`` on all other architectures.
+
+  * - ``MIGRAPHX_DISABLE_LAYERNORM_FUSION``
+    - Disables layernorm fusion.
+      | ``0``: Layernorm fusion is enabled.
+      | ``1``: Layernorm fusion is disabled.
+      | Default: ``0``
+  
+  * - ``MIGRAPHX_DISABLE_MIOPEN_POOLING``   
+    - When set to ``1``, MIOpen pooling is disabled and MIGraphX pooling is enabled.
+      | ``0``: MIOpen pooling is enabled.
+      | ``1``: MIGraphX pooling is enabled.
+      | Default: ``0``
+
+  * - ``MIGRAPHX_USE_FAST_SOFTMAX``
+    - Enables fast softmax optimization to speed up softmax computations.
+      | 0: Softmax optimization isn't used.
+      | 1: Softmax optimization is used.
+      | Default: 0
+
+  * - ``MIGRAPHX_MLIR_USE_SPECIFIC_OPS``

```

Comment:
The options here are: 
`attention` : Enable attention fusion (enabled by default on MI300, required for other archs)
`convolution` : Use MLIR generated kernels for all convolutions (default behavior can pick MIOpen for some cases)
`dot` : Use MLIR generated kernels for all GEMMs (default behavior is to use hipblaslt)
`fused_convolution` : Use MLIR generated kernels for convolutions when doing so enables extra fusions
`fused_dot` : Use MLIR generated kernels for GEMMs when doing so enables extra fusions
`fused` : Equivalent of setting both fused_dot and fused_convolution? @pfultz2 please confirm

Additionally, the `~` character can be used to prefix the above tokens to achieve the inverse behavior. Eg:
`~convolution` : Never use MLIR generated kernels for convolution